### DISCUSSION TITLE
How do I use the summary module?

### DISCUSSION THREAD

**Discussion by cvieiraneto** [explosion/spacy-llm#298] [General]
I'm trying to use the summary pipe, but I haven't found any references for how to use it. Does anyone know?

```python
nlp = spacy.blank("pt")
llm = nlp.add_pipe("llm_summarization")
nlp("""The United Nations, referred to informally as the UN, is an
    intergovernmental organization whose stated purposes are  to maintain
    international peace and security, develop friendly relations among nations,
    achieve international cooperation, and serve as a centre for harmonizing the
    actions of nations. It is the world's largest international organization.
    The UN is headquartered on international territory in New York City, and the
    organization has other offices in Geneva, Nairobi, Vienna, and The Hague,
    where the International Court of Justice is headquartered.\n\n The UN was
    established after World War II with the aim of preventing future world wars,
    and succeeded the League of  Nations, which was characterized as
    ineffective.""")
```    

output: same text (without resume)

**Comment by tleers** [comment]
I ran into the same issue. For whatever reason, the output of the summariser is nested somewhere in a dictionary under the attribute "user_data". 

E.g.:

```
doc = nlp("text to summarize")
doc.user_data
```

**Comment by rmitsch** [comment]
To expand on this: spaCy pipeline components don't modify the text in place. Doing that for this particular `spacy-llm` task would have run counter to this convention.

By default the summary is stored in `doc.user_data["summary"]`. It can be stored in an arbitrary field in `user_data` by setting the `field` argument in the config.

The [documentation](https://spacy.io/api/large-language-models#summarization-v1) mentions this, but perhaps we should highlight this more/add some explanation as to why the `Doc` instance's text isn't modified directly.

**Reply by tleers** [reply]
In the documentation you linked, I see no mention of this. On top of that, when I use the following code:
```
doc = nlp("Summarize this text please.")
doc.user_data["summary"]
```

just results in`KeyError: 'summary'`.

The user_data object looks like this: 
`{('._.', 'summary', None, None): 'The text is too short to be summarized.'}`

Am I doing something wrong, @rmitsch?

**Reply by rmitsch** [reply]
See here:
![image](https://github.com/explosion/spacy-llm/assets/7152225/a5c039ab-4e22-441b-89dd-3bad7a591154)

**Reply by rmitsch** [reply]
I. e. the summary is, by default, available in `doc._.summary`.

**Comment by rmitsch** [comment]
I'll convert this to a discussion, as this is not an issue.
